### ISSUE TITLE
Report ""Use immersive mode for image gallery" makes the layout go up then down"

### ISSUE THREAD

**Issue by K1rakishou** [Adamantcheese/Kuroba#1183]
UUID: a59ae24188c577d92e7d7751dd407a58
Build flavor: dev
Version name: v4.13.6-6ddbf11
OS info: Android 10, sdk version: 29
Report description: You probably have noticed this already, that's the only app I've seen this happening and it's kinda weird. KurobaEX doesn't have this issue, so is it possible to backport the code?

**Comment by K1rakishou** [comment]
Just FYI in KurobaEx I use edge-to-edge and apparently using edge-to-edge and SYSTEM_UI_FLAG_LAYOUT_STABLE flag in the same activity is bugged on Android 9 and below. What happens is after switching into fullscreen mode and the back all animations (and onLayout calls) are breaking until the app restart. I use a hack to avoid this on Android 9 and below and it has the same "jumping" effect. Works fine on Android 10+ though. The only way to avoid this is apparently to use a separate activity for image viewer and only use fullscreen mode in that activity, not the main one.

**Comment by Adamantcheese** [comment]
Yeah, it's the same thing from #195 and #338. Downsides of not using Android's activity system I guess.
